wikimedia/wikipedia-ios
# Show proper navigation bar for references

**Fixes Phabricator ticket:** https://phabricator.wikimedia.org/T268586

### Notes
* After lots of research, the fix ended up being simple: `ReferenceViewController` is never displayed as a push, and its nav bar should know it is displayed modally. (While not actually a modal presentation, it's the best choice of those options - and naturally causes the proper nav bar behavior.)
* Bug would have snuck in [this PR](https://github.com/wikimedia/wikipedia-ios/pull/3640/files), as we changed how we're assembling the `navigationItem` stack.

### Test Steps
1. Open app to an article, scroll to Reference section. Tap a reference number. App should scroll up to that reference, and have a new reference toolbar and nav bar. 
1. When you tap anywhere on the article (or the x in the top right corner), the nav bar and toolbar should go back to normal.
1. Confirm it works on iPad, iPhone 8, and iPhone X-sized devices.

### Screenshots/Videos 
<img width="197" alt="Screenshot 8" src="https://user-images.githubusercontent.com/9295855/100947130-d4e05d80-34b9-11eb-8fe3-af33773d45de.png"><img width="298" alt="Screenshot 7" src="https://user-images.githubusercontent.com/9295855/100947121-d27e0380-34b9-11eb-881b-bcd503e899a4.png">
